File: malcompm_2025_midyear_ece_2.java
//Final Step. Development Procedure
/*
Steps 1 - 3: Thonny IDE and Postman IDE
Steps 4 - 9: Android Studio IDE
*/

/*
4. Download the provided ZIP file, extract in a known location.
4.1. Open Android Studio IDE, go to File Menu >> Open and navigate to the location of the extracted file.
4.2. Wait for Sync to finish, then run AVD. Report encountered errors.
4.3. No error? Proceed to next steps.
*/

/*
5. Before <application> tag, add the following line of code in Android Manifest.xml
    <uses-permission android:name="android.permission.INTERNET"/>
*/

/*
6. After onCreate method of MainActivity.java, create the following user-define methods.

6.1. para sa widgets (components).
    void pinControl() {
        // wala muna laman
    }

6.2. para sa HTTP request (JSON).
    void sendJsonPostRequest() {
        // wala muna laman
    }

6.3. No error? Proceed to next steps.
*/

/*
// 7. Copy the contents of the user-define methods.
*/

/*
// 7.1. Copy the contents of the user-define methods for pinControl().
*/

    void pinControl() {
        ImageView iv_01_pin_on = findViewById(R.id.iv_01_pin_on);
        ImageView iv_02_pin_off = findViewById(R.id.iv_02_pin_off);

        iv_01_pin_on.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getBaseContext(), "Turning On...", Toast.LENGTH_SHORT).show();
                sendJsonPostRequest("On");
            }
        });

        iv_02_pin_off.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getBaseContext(), "Turning On...", Toast.LENGTH_SHORT).show();
                sendJsonPostRequest("Off");
            }
        });
    }

/*
// 7.2. Copy the contents of the user-define methods for sendJsonPostRequest().
7.2.1. Change the ip address to the IP of your Pico W.
*/

    private void sendJsonPostRequest(String s) {
        try {
            RequestQueue requestQueue = Volley.newRequestQueue(this);
            JSONObject jsonBody = new JSONObject();
            //{"ledInternal": 1, "ledExternal": 0,}
            jsonBody.put("led_internal_main", s);
            jsonBody.put("ledExternal", s);

            final String mRequestBody = jsonBody.toString();

            StringRequest stringRequest = new StringRequest(Request.Method.POST, "http://192.168.8.100/api/pagsend-ng-tatlong-data", new Response.Listener<String>() {
                @Override
                public void onResponse(String response) {
                    Log.i("MPM_LOG", response);
                }
            }, new Response.ErrorListener() {
                @Override
                public void onErrorResponse(VolleyError error) {
                    Log.e("LOG_RESPONSE", error.toString());
                }
            }) {
                @Override
                public String getBodyContentType() {
                    return "application/json; charset=utf-8";
                }

                @Override
                public byte[] getBody() throws AuthFailureError {
                    try {
                        return mRequestBody.getBytes("utf-8");
                    } catch (UnsupportedEncodingException uee) {
                        VolleyLog.wtf("Unsupported encoding while trying to get the bytes of %s using %s", mRequestBody, "utf-8");
                        return null;
                    }
                }

                @Override
                protected Response<String> parseNetworkResponse(NetworkResponse response) {
                    String responseString = "";
                    if (response != null) {
                        responseString = String.valueOf(response.statusCode);
                    }
                    return Response.success(responseString, HttpHeaderParser.parseCacheHeaders(response));
                }
            };
            requestQueue.add(stringRequest);
            // changing progress bar visibility on below line.
            // mpm_loading_progress_bar.setVisibility(View.GONE);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

/*
7.3. No error? Proceed to next steps.
*/

/*
8. Run AVD, click ON and OFF icon.
8.1. Observe Thonny Shell responses. No error? Proceed to next steps.
8.2. Modify the sendJsonPostRequest(String s) method to include the correct variable names used in python code.
8.2.1. Change jsonBody.put("led_internal_main", s); to >>
8.2.2. Change jsonBody.put("ledExternal", s); to >>
8.2.3. Add jsonBody.put("variable-name-na-nasa-python", s);

8.3. Run AVD, click ON and OFF icon, built-in LED must turn on and off.
*/

File: malcompm_2025_midyear_ece_2.py
#Final Step. Development Procedure
"""
Steps 1 - 3: Thonny IDE and Postman IDE
Steps 4 - 6: Android Studio IDE
"""

#1.	Open your code in Thonny IDE, and “Save As” main.py in your Pico W.
#2.	Modify the python code related to the following block of codes …

WIFI_SSID = "HUAWEI-411C"
WIFI_PASS = "a0yy8334"

ip = connect_to_wifi(WIFI_SSID, WIFI_PASS)
kimi_no_nawa = "MP MALCO"
print("\n")
print("Performer:", kimi_no_nawa)
if(ip is not None):
    print("\nConnected ka na sa network!")
    print("Ang iyong IP address ay:", ip, "\n")
else:
    print("\nDi konektado!\n")
    while(ip is None):
        print("\nKonekting ulit...\n")
        ip = connect_to_wifi(WIFI_SSID, WIFI_PASS)
        delay(3)

#3.	Add the new block of codes before server.run() and test via Postman IDE.

hw_main = led_internal_main
""" dataMain, dataUna, dataPangalawa"""
@server.route("/api/pagsend-ng-tatlong-data", methods=["POST"])
def dalawangDataSignal(request):
    print("\nHTTP POST request received.")
    
    try:
        aking_data_main = request.data["dataMain"]
        print("\tPico received data (dataMain):", aking_data_main)
        if(aking_data_main == "On"):
            hw_main.value(1)
        else:
            hw_main.value(0)
    except:
        print("\tPico received data (dataMain):", "Wala talaga, as in wala.")

    try:
        aking_data_una=request.data["dataUna"]
        print("\tPico received data (dataUna):", aking_data_una)
        if(aking_data_una == "On"):
            hw_uno.value(1)
        else:
            hw_uno.value(0)
    except:
        print("\tPico received data (dataUna):", "Wala talaga, as in wala.")

    try:
        aking_data_pangalawa=request.data["dataPangalawa"]
        print("\tPico received data (main):", aking_data_pangalawa)
        if(aking_data_pangalawa == "On"):
            hw_dos.value(1)
        else:
            hw_dos.value(0)
    except:
        print("\tPico received data (dataPangalawa):", "Wala talaga, as in wala.")
        
        #return json.dumps({"message" : "Command sent successfully!"}), 200, {"Content-Type": "application/json"}
    
    diksyonaryo = dict()
    try:
        diksyonaryo["msg_"]= "Command sent successfully!"
        diksyonaryo["msg_0"]= aking_data_main
        diksyonaryo["msg_1"]= aking_data_una
        diksyonaryo["msg_2"]= aking_data_pangalawa
    except:
        diksyonaryo["msg_"]= "Command not processed!"
        
    print("\nSending response.", diksyonaryo, "\n")
        
    return json.dumps(diksyonaryo), 200, {"Content-Type": "application/json"}
